docs/learn/learn-comparisons.md

```diff
@@ -1,63 +1,212 @@
+| **State Transition Logic**   | General-purpose but constrained by zk-circuit implementation complexity.                                                                                                                                                                                       | Can only support EVM state transitions with assumptions of optimistic validity since they have to be checked in the EVM environment<sup>1</sup>.                             | Each parachain can define its unique state transition function (STF), compiled to Wasm and validated per Parachain Protocol rules.                                                                                                                                                                                                                  |
+| **Development Complexity**   | Complex due to the mathematics of zero-knowledge proofs.                                                                                                                                                                                                       | Moderate complexity, requiring fraud-proof implementation.                                                                                                                   | Moderate to high complexity; parachain runtimes must be written in Wasm-compatible languages but can define custom logic and governance. Parachain maintainance can be an overhead.                                                                                                                                                                 |
+| **Data Availability**        | Requires posting minimal data proofs to L1.                                                                                                                                                                                                                    | Posts complete data on-chain during challenge periods, increasing cost.                                                                                                      | Built-in data availability with validators ensuring distributed state storage and reconstruction in case of disputes.                                                                                                                                                                                                                               |
```

Comment:
Data availability requirements posted by the optimistic and ZK rollups is precisely the same.